captum/insights/frontend/src/components/FilterContainer.tsx

```diff
@@ -28,34 +43,31 @@ class FilterContainer extends React.Component {
     };
   }
 
-  handleClassDelete = (i) => {
+  handleClassDelete = (classId: number) => {
     const classes = this.state.classes.slice(0);
-    const removed_class = classes.splice(i, 1);
-    const suggested_classes = [].concat(
-      this.state.suggested_classes,
-      removed_class
-    );
+    const removed_class = classes.splice(classId, 1);
+    const suggested_classes = [...this.state.suggested_classes, ...removed_class];
     this.setState({ classes, suggested_classes });
   };
 
-  handleClassAdd = (added_class) => {
-    const classes = [].concat(this.state.classes, added_class);
+  handleClassAdd = (added_class: TagClass) => {
+    const classes = [...this.state.classes, added_class];
     const suggested_classes = this.state.suggested_classes.filter(
       (t) => t.id !== added_class.id
     );
     this.setState({ classes, suggested_classes });
   };
 
-  handleInputChange = (event) => {
+  handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
     const target = event.target;
     const value = parseEventTargetValue(event.target);
     const name = target.name;
     this.setState({
       [name]: value,
-    });
+    } as any);
   };
 
-  handleArgumentChange = (event) => {
+  handleArgumentChange = (event: React.ChangeEvent<HTMLInputElement>) => {
```

Comment:
Should this be `React.ChangeEvent<HTMLInputElement | HTMLSelectElement>`? You can also create a new type using lookup types so the code doesn't have to be repeated.

Reply:
Added union type for Input and Select.